### PR TITLE
Made examples a package

### PR DESCRIPTION
**Pull request by brendandixon**
@ahmetalpbalkan This should fix the CI build.

Signed-off-by: Brendan Dixon <brendand@microsoft.com>

### CODE REVIEW

File: arm/main.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import (
+	"fmt"
+	"os"
+
+	"github.com/azure/azure-sdk-for-go/arm/examples"
+)
+
+func help() {
+	fmt.Println(`Select an example to run:
+		Client  -- Basic storage account example
```
Comment (ahmetalpbalkan) [inline arm/main.go]: Never seen a subcommand that starts with an uppercase letter. :-)

File: arm/examples/client.go [hunk 2]
```diff
@@ -32,17 +32,17 @@ func (i Inspectors) ByInspecting() autorest.RespondDecorator {
 	}
 }
 
-func main() {
-	if len(os.Args) < 2 {
+func Client(args []string) {
+	if len(args) < 1 {
 		fmt.Println("Please provide a name to check")
-		os.Exit(1)
+		return
```
Comment (ahmetalpbalkan) [inline arm/examples/client.go]: Use log.Fatalf instead of fmt.Print+return

**Comment by ahmetalpbalkan** [comment]
Lgtm. Feel free to merge once you fix the comment above if it makes sense. Log.fatal is mode idiomatic although it doesnt give the same execution flow.

File: arm/examples/client.go [hunk 3]
```diff
@@ -32,17 +32,17 @@ func (i Inspectors) ByInspecting() autorest.RespondDecorator {
...
 }
 
-func main() {
-	if len(os.Args) < 2 {
-		fmt.Println("Please provide a name to check")
-		os.Exit(1)
+func Client(args []string) {
+	if len(args) < 1 {
+		log.Fatalf("Please provide a name to check")
+		return
```
Comment (ahmetalpbalkan) [inline arm/examples/client.go]: You don't need return anymore it already will exit

File: arm/examples/client.go [hunk 4]
```diff
@@ -50,7 +49,7 @@ func main() {
 	spt, err := helpers.NewServicePrincipalTokenFromCredentials(c, azure.AzureResourceManagerScope)
 	if err != nil {
 		fmt.Println(err)
```
Comment (ahmetalpbalkan) [inline arm/examples/client.go]: Ideally you should never call fmt.println in this pkg for errors as it's a very short program and exits on every error. Just log.fatal

File: arm/examples/storage.go [hunk 5]
```diff
@@ -78,13 +74,13 @@ func main() {
 	if err != nil {
 		if sa.Response.StatusCode != 202 {
```
Comment (ahmetalpbalkan) [inline arm/examples/storage.go]: Use http.StatusWhatever. I don't know what 202 means :-)

**Comment by ahmetalpbalkan** [comment]
Also spotted some struct initializations in storage example where you could have used struct initializer instead of assignment. Can't comment at that point as github mobile sucks.

File: arm/examples/storage.go [hunk 6]
```diff
@@ -78,13 +74,13 @@ func main() {
...
 			fmt.Printf("Creation of %s.%s failed with err -- %v\n", resourceGroup, name, err)
-			os.Exit(1)
+			return
 		} else {
 			fmt.Printf("Create initiated for %s.%s -- poll %s to check status\n",
```
Comment (ahmetalpbalkan) [inline arm/examples/storage.go]: %s/%s is more accurate than %s.%s but fwiw.

Comment (brendandixon) [inline arm/examples/storage.go]: @ahmetalpbalkan I'll change it later...

**Comment by brendandixon** [comment]
@ahmetalpbalkan Regarding struct initialization. Point me to specifics and I'll fix later.

**Comment by ahmetalpbalkan** [comment]
Just search for {} in storage.go

**Comment by brendandixon** [comment]
@ahmetalpbalkan Yah, right. Searching for {} returns all those interfaces as well. :) I'll find / fix later.

**Comment by ahmetalpbalkan** [comment]
You have interfaces in arm/examples?

**Comment by brendandixon** [comment]
@ahmetalpbalkan Oh, I thought you meant in arm generally. No worries. I'll fix them.

**Pull request merged**